README.tfdoc.hcl

```diff
@@ -0,0 +1,1223 @@
+                END
+              }
+            }
+          }
+
+          attribute "redirect" {
+            type        = object(redirect)
+            description = <<-END
+              Configuration block for creating a redirect action. Required if
+              `type` is `redirect`.
+            END
+
+            attribute "status_code" {
+              type        = string
+              default     = "HTTP_302"
+              description = <<-END
+                HTTP redirect code. The redirect is either permanent (`HTTP_301`)
+                or temporary (`HTTP_302`).
+              END
+            }
+
+            attribute "host" {
+              type        = string
+              default     = "HTTP_302"
+              description = <<-END
+                Hostname. This component is not percent-encoded. The hostname can
+                contain `#{host}`.
+              END
+            }
+
+            attribute "path" {
+              type        = string
+              default     = "#{host}"
+              description = <<-END
+                Absolute path, starting with the leading `"/"`. This component is
+                not percent-encoded. The path can contain `#{host}`, `#{path}`,
+                and `#{port}`.
+              END
+            }
+
+            attribute "port" {
+              type        = string
+              default     = "#{port}"
+              description = <<-END
+                The port. Specify a value from `1` to `65535` or `#{port}`.
+              END
+            }
+
+            attribute "protocol" {
+              type        = string
+              default     = "#{protocol}"
+              description = <<-END
+                The protocol. Valid values are `HTTP`, `HTTPS`, or `#{protocol}`. 
+              END
+            }
+
+            attribute "query" {
+              type        = string
+              default     = "#{query"
+              description = <<-END
+                The query parameters, URL-encoded when necessary, but not
+                percent-encoded. Do not include the leading "?".
+              END
+            }
+          }
+        }
+
+        attribute "conditions" {
+          required    = true
+          type        = set(conditions)
+          description = <<-END
+            A Condition block. Multiple condition blocks of different types can
+            be set and all must be satisfied for the rule to match.
+          END
+
+          attribute "host_header" {
+            type        = object(host_header)
+            default     = []
+            description = <<-END
+              The maximum size of each pattern is 128 characters. Comparison is
+              case insensitive. Wildcard characters supported: `*` (matches 0 or
+              more characters) and `?` (matches exactly 1 character). Only one
+              pattern needs to match for the condition to be satisfied.
+            END
+
```

Comment:
`http_header_name` seems to be missing https://github.com/mineiros-io/terraform-aws-lb-listener/blob/main/variables.tf#L251